domain/application/service_test.go

```diff
@@ -122,8 +123,13 @@ func (s *serviceSuite) createSecrets(c *gc.C, appName, unitName string) (appSecr
 		Data:       coresecrets.SecretData{"foo": "bar"},
 		RevisionID: ptr(uuid.MustNewUUID().String()),
 	}
-	err := s.secretState.CreateCharmApplicationSecret(ctx, 1, appSecretURI, appName, sp)
-	c.Assert(err, jc.ErrorIsNil)
+	_ = s.secretState.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
+		appUUID, err := s.secretState.GetApplicationUUID(ctx, appName)
```

Comment:
An improvement here is that we could pass in the app uuid since we would have it when the app is created. But this works as well for now.